schema-lending-extended.graphql

```diff
@@ -0,0 +1,1173 @@
+# different duration/maturity per market. You can append a counter
+# to the IDs to differentiate.
+type InterestRate @entity {
+  " { Interest rate side }-{ Interest rate type }-{ Market ID } "
+  id: ID!
+
+  " Interest rate in percentage APY. E.g. 5.21% should be stored as 5.21 "
+  rate: BigDecimal!
+
+  " Duration of the loan in days. Only applies to fixed term lending (e.g. Notional) "
+  duration: Int
+
+  " Maturity of the loan in block height. Only applies to fixed term lending (e.g. Notional) "
+  maturityBlock: BigInt
+
+  " The party the interest is paid to / received from "
+  side: InterestRateSide!
+
+  " The type of interest rate (e.g. stable, fixed, variable, etc) "
+  type: InterestRateType!
+}
+
+#############################
+##### Protocol Metadata #####
+#############################
+
+interface Protocol {
+  " Smart contract address of the protocol's main contract (Factory, Registry, etc) "
+  id: ID!
+
+  " Name of the protocol, including version. e.g. Uniswap v3 "
+  name: String!
+
+  " Slug of protocol, including version. e.g. uniswap-v3 "
+  slug: String!
+
+  " Version of the subgraph schema, in SemVer format (e.g. 1.0.0) "
+  schemaVersion: String!
+
+  " Version of the subgraph implementation, in SemVer format (e.g. 1.0.0) "
+  subgraphVersion: String!
+
+  " Version of the methodology used to compute metrics, loosely based on SemVer format (e.g. 1.0.0) "
+  methodologyVersion: String!
+
+  " The blockchain network this subgraph is indexing on "
+  network: Network!
+
+  " The type of protocol (e.g. DEX, Lending, Yield, etc) "
+  type: ProtocolType!
+
+  ##### Quantitative Data #####
+
+  " Current TVL (Total Value Locked) of the entire protocol "
+  totalValueLockedUSD: BigDecimal!
+
+  " Current PCV (Protocol Controlled Value). Only relevant for protocols with PCV. "
+  protocolControlledValueUSD: BigDecimal
+
+  " Number of cumulative unique users "
+  cumulativeUniqueUsers: Int!
+
+  " Revenue claimed by suppliers to the protocol. LPs on DEXs (e.g. 0.25% of the swap fee in Sushiswap). Depositors on Lending Protocols. NFT sellers on OpenSea. "
+  cumulativeSupplySideRevenueUSD: BigDecimal!
+
+  " Gross revenue for the protocol (revenue claimed by protocol). Examples: AMM protocol fee (Sushi’s 0.05%). OpenSea 10% sell fee. "
+  cumulativeProtocolSideRevenueUSD: BigDecimal!
+
+  " All revenue generated by the protocol. e.g. 0.30% of swap fee in Sushiswap, all yield generated by Yearn. "
+  cumulativeTotalRevenueUSD: BigDecimal!
+
+  " Total number of pools "
+  totalPoolCount: Int!
+
+  ##### Snapshots #####
+
+  " Daily usage metrics for this protocol "
+  dailyUsageMetrics: [UsageMetricsDailySnapshot!]! @derivedFrom(field: "protocol")
+
+  " Hourly usage metrics for this protocol "
+  hourlyUsageMetrics: [UsageMetricsHourlySnapshot!]! @derivedFrom(field: "protocol")
+
+  " Daily financial metrics for this protocol "
+  financialMetrics: [FinancialsDailySnapshot!]! @derivedFrom(field: "protocol")
```

Comment:
Should we make this `dailyFinancialMetrics` for consistency with `dailyUsage Metrics` ?

Reply:
hmm, good point. There are some historical reasons to this. We only had daily values before, so they were named `usageMetrics` and `financialMetrics`. Later people wanted hourly data for usage, so I made `usageMetrics` into daily and hourly. But no one asked for hourly financial metrics so I left as is